docs/types/dynamic-dispatch.md

```diff
@@ -162,3 +162,159 @@ main = Test.simplify
+  - Builtin types are usually implemented in the engine, and not with pure Enso
+    code.
+- **parent type** of a type `My_Type` is a type that `My_Type` "extends", i.e.
+  `My_Type` inherits all the methods defined on its parent type.
+  - Every type has exactly one parent type, except for `Any` type.
+  - `Any` has no parent type.
+  - All types have implicit parent type `Any`.
+  - There are some exceptions for some _builtin types_
+    - For example `Float` and `Integer` builtin types have `Number` parent.
+    - `Number` has `Any` parent.
+- **symbol table**.
+  - Every type has an associated symbol table.
+  - Symbol table maps symbols to their definitions.
+  - All definitions are methods.
+    - Atom constructors are methods.
+    - Atom fields are methods. More specifically, every atom field has an
+      associated getter method.
+
+This section describes the \_method invocation process, which resolves a
+concrete method definition for a concrete call site and evaluates it. For a
+method call expression `Receiver.symbol`, this section focuses only on a single
+dispatch based on `Receiver` argument. For multiple dispatch, see the
+[Multiple Dispatch](#multiple-dispatch) section.
+
+Method resolution algorithm for `Receiver.symbol` first determines the _type_ of
+the `Receiver`, and then finds the method definition in its _symbol table_:
+
+1. **Determine the type of `Receiver`:**
+
+- 1.1. If `Receiver` is type, the result will be _eigen type_
+- 1.2. If `Receiver` is a value (instance / atom), the result will be the _type
+  of the value_
+- 1.3. If `Receiver` is a module, the result will be the _associated type_ for
+  the module
+- 1.4. If `Receiver` is a polyglot object, method resolution and invocation will
+  be handled according to the [polyglot interoperability](../polyglot/README.md)
+  rules.
+  - Polyglot object can be, for example:
+    - A Java class, imported by `polyglot java import ...` statement.
+    - Java object instance, created by `Java_Class.new ...` expression.
+    - Javascript, Python, or any other allowed foreign language object returned
+      by a foreign method call.
+    - Refer to [polyglot readme](../polyglot/README.md) for more details.
+- 1.5. If there is no `Receiver`, we are just looking for a variable in the
+  current lexical scope or any parent scopes. See
+  [Lexical scope lookup](#lexical-scope-lookup).
+
+2. **Look up symbol in the symbol table of the determined type:**
+
+- 2.1. Lookup the `symbol` in the Receiver's type and all its parent types.
+- 2.2. If it is found, continue to 3.
+- 2.3. If it is not found, continue to 4.
+
+3. **Invoke the method with defaulted self**:
+
+- 3.1. `symbol` is a method in Receiver's type (or its parent type) symbol
+  table.
+- 3.2. Such method is treated as if it's first parameter is named `self` and has
+  the default value of `Receiver`.
+- 3.3. The call site can specify named `self` argument
+
+4. **Static invocation without bound self**:
```

Comment:
If the first argument of the method is `self=`, then it is static method invocation syntax.